clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -783,7 +784,13 @@ public void unsubscribe(TopicPartition... partitions) {
     private void scheduleAutoCommitTask(final long interval) {
         DelayedTask task = new DelayedTask() {
             public void run(long now) {
-                commit(CommitType.ASYNC);
+                commitAsync(new OffsetCommitCallback() {
```

Comment:
I thought about whether we need to create a new callback for each commit, but since auto commit is not frequent and the callback is very small, it might be fine. If you think it matters we can reuse one callback.